helion/autotuner/config_generation.py

```diff
@@ -199,6 +199,6 @@ def encode_config(self, flat_config: FlatConfig) -> list[float]:
 
         for flat_idx, spec in enumerate(self.flat_spec):
             value = flat_config[flat_idx]
-            encoded.append(spec.encode_scalar(value))
+            encoded.extend(spec.encode(value))
```

Comment:
Let's add some asserts here the `spec.dim` matches the length of the return value of encode.

I'm worried we will have a bug where those two don't line up.